src/main/java/core/Parser.java

```diff
@@ -0,0 +1,191 @@
+package core;
+
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+
+import command.AddDeadlineCommand;
+import command.AddEventCommand;
+import command.AddRecurringCommand;
+import command.AddToDoCommand;
+import command.Command;
+import command.DeleteCommand;
+import command.DoneCommand;
+import command.ExitCommand;
+import command.FindCommand;
+import command.ListCommand;
+import command.ResetCommand;
+import dukexception.DukeException;
+
+
+/**
+ * Parser the input from user to Command that can execute.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input to recognisable command.
+     *
+     * @param userInput input from user.
+     * @return command to be executed.
+     * @throws DukeException user input is incomplete or invalid.
+     */
+    public Command parse(String userInput) throws DukeException {
+
+        if (userInput.contains(" ")) {
+            String keyword = userInput.substring(0, userInput.indexOf(" "));
+            String information = userInput.substring(userInput.indexOf(" ") + 1);
+
+            switch (keyword) {
+            case "done":
+                return parseDone(information);
+            case "delete":
+                return parseDelete(information);
+            case "todo":
+                return parseTodo(information);
+            case "deadline":
+                return parseDeadline(information);
+            case "event":
+                return parseEvent(information);
+            case "recurring":
+                return parseRecurring(information);
+            case "find":
+                return parseFind(information);
+            default:
+                throw new DukeException(ErrorMessage.UNKNOWN_INPUT.toString());
+            }
+        } else {
+            switch (userInput) {
+            case "list":
+                return parseList();
+            case "bye":
+                return parseBye();
+            case "reset":
+                return parseReset();
+            case "done":
```

Comment:
Maybe all these empty case scenarios are unnecessary since they'll just keep moving on? Just a suggestion to shorten this part a little.